MoneySplitter.Win10/Common/CollaboratorModelFactory.cs

```diff
@@ -0,0 +1,130 @@
+        }
+
+        public IEnumerable<CollaboratorModel> GetDebtors()
+        {
+            return _transactionsManager.UserTransactions.Where(tr => tr.Owner.Id == _membershipService.CurrentUser.Id)
+                .SelectMany(tr => ConvertTransactionModelToCollabaratorModel(tr, CollaboratorStatus.ONE_DEBT))
+                .GroupBy
+                (
+                    cl => cl.Email,
+                    cl => cl,
+                    (key, value) => new { Email = key, CollaboratorRecords = value }
+                )
+                .Select(cl => ConvertCollaboratorRecordsToOneRecord(cl.CollaboratorRecords, CollaboratorStatus.MANY_DEBT));
+        }
+
+        public IEnumerable<CollaboratorModel> GetLendPersons()
+        {
+            return _transactionsManager.UserTransactions.Where(tr => tr.Owner.Id != _membershipService.CurrentUser.Id)
+                .Select(
+                            tr => ConvertDataToCollabatorModel(tr.Owner, tr, CollaboratorStatus.ONE_LEND,
+                            GetTransactionStatus(_membershipService.CurrentUser, tr))
+                       )
+               .GroupBy
+                (
+                    cl => cl.Email,
+                    cl => cl,
+                    (key, value) => new { Email = key, CollaboratorRecords = value }
+                )
+                .Select(cl => ConvertCollaboratorRecordsToOneRecord(cl.CollaboratorRecords, CollaboratorStatus.MANY_LEND));
+        }
+
+        private IEnumerable<CollaboratorModel> ConvertTransactionModelToCollabaratorModel(TransactionModel transactionModel, CollaboratorStatus collabaratorStatus)
+        {
+            return transactionModel.Collaborators.Where(cl => cl.Id != _membershipService.CurrentUser.Id)
+                     .Select(cl => ConvertDataToCollabatorModel(cl, transactionModel, collabaratorStatus, TransactionStatus.IN_BEGIN))
+                     .Concat(
+                               transactionModel.InProgressIds.Where(id => id != _membershipService.CurrentUser.Id)
+                               .Select(
+                                       userId => ConvertDataToCollabatorModel(
+                                                   GetCollaborator(userId, transactionModel), 
+                                                   transactionModel, collabaratorStatus, TransactionStatus.IN_PROGRESS
+                                                 )     
+                                      )
+                            );
+        }
+
+        private CollaboratorModel ConvertDataToCollabatorModel(
+            UserModel userModel,
+            TransactionModel transactionModel,
+            CollaboratorStatus collaboratorStatus,
+            TransactionStatus transactionStatus)
+        {
+            return new CollaboratorModel
+            {
+                FullName = userModel.Name + " " + userModel.Surname,
+                Cost = Math.Round(transactionModel.SingleCost, 2),
```

Comment:
push 2 to defines or make it const in this class

Reply:
fixed